src/assets/js/alma-widgets-inject.js

```diff
@@ -0,0 +1,173 @@
+/**
+ * Inject payment plan widget.
+ *
+ * @package Alma_Gateway_For_Woocommerce
+ */
+
+jQuery( document ).ready(
+	function ( $ ) {
+
+		var almaWidgetHelper = new AlmaWidgetHelper();
+		var settings         = almaWidgetHelper.getSettings();
+		if ( ! settings ) {
+			return;
+		}
+
+		var jqueryUpdateEvent = settings.jqueryUpdateEvent;
+		var firstRender       = settings.firstRender;
+
+		if ( firstRender ) {
+			almaWidgetHelper.init();
+		}
+
+		if ( jqueryUpdateEvent ) {
+			$( document.body ).on(
+				jqueryUpdateEvent,
+				function () {
+					// WooCommerce animates the appearing of the product's price when necessary options have been selected,
+					// or its disappearing when some choices are missing. We first try to find an ongoing animation to
+					// update our widget *after* the animation has taken place, so that it uses up-to-date information/DOM
+					// in init.
+					var amountElement = almaWidgetHelper.getAmountElement( settings )
+					var timer         = $.timers.find(
+						function ( t ) {
+							return t.elem === jQuery( amountElement ).closest( '.woocommerce-variation' ).get( 0 );
+						}
+					)
+
+					if (timer) {
+						window.setTimeout( almaWidgetHelper.init, timer.anim.duration )
+					} else if ( almaWidgetHelper.isVisible( amountElement ) || ! settings.amountQuerySelector ) {
+						almaWidgetHelper.init();
+					}
+				}
+			);
+		}
+	}
+);
+
+/**
+ * Alma admin General Helpers.
+ */
+function AlmaWidgetHelper() {
```

Comment:
Put this function in a new file: AlmaWidgetHelper.js

Reply:
Noté pour plus tard !